test/Chainweb/Test/Pact/Utils.hs

```diff
@@ -428,8 +440,12 @@ withPactCtxSQLite v cutDB bhdbIO pdbIO f =
       pdb <- pdbIO
       (_,s) <- ios
       (dbSt, cpe) <- initRelationalCheckpointer' initBlockState s logger gasEnv
+      adb <- newMVar Nothing
       !ctx <- TestPactCtx
         <$!> newMVar (PactServiceState Nothing)
-        <*> pure (PactServiceEnv Nothing cpe spv pd pdb bhdb)
-      evalPactServiceM ctx (initialPayloadState v cid mempty)
+        <*> pure (PactServiceEnv Nothing cpe spv pd pdb bhdb adb)
+      evalPactServiceM ctx (initialPayloadState v cid)
       return (ctx, dbSt)
+
+withMVarResource :: a -> (IO (MVar a) -> TestTree) -> TestTree
+withMVarResource value = withResource (newMVar value) (const $ return ())
```

Comment:
This seems useful more generally. Maybe it can be moved to `Chainweb.Test.Utils`, so that it is available to other tests, too?

Reply:
Maybe but I'm not doing it before I go on vacation :) :)